### ISSUE TITLE
Multiple classifications in Viewer

### ISSUE THREAD

**Issue by pwalsh** [openspending/openspending#1315]
# Description

In discussion with the Mexican Ministry of Finance, a point was raised to me around a use case for filtering on multiple classifications. Currently, the use case they have is not possible in the Viewer, so they use the API directly and build their own visualisation.

I want to understand the use case and see if it can be elevated to a feature of the Viewer.

Lorena from the team at Mexico agreed to send me an email with details.

**Comment by pwalsh** [comment]
Still waiting on more information. Keeping open on the backlog.

**Comment by pwalsh** [comment]
Explanation from Lorena:

> We have a variable named "Ramo" which belongs in both the administrative and the activity classification. "Ramo" in itself is administrative concept since it explains "who" is spending the budget; however, it is also necessary to use it in the Activity classification in order to have a unique identifier for Budget Program; for example, the Modalidad "S" with Programa Presupuestario "72" is different depending if it belongs to Ramo "11", "12" or "20". As you correctly said, the API allows us to make queries taking this into account: (for example http://next.openspending.org/api/3/cubes/6018ab87076187018fc29c94a68a3cd2:presupuesto-mexico-2008-2018/facts/?cut=date_2.CICLO:%222018%22|administrative_classification_2.ID_RAMO:%2220%22|activity_ID_MODALIDAD.ID_MODALIDAD:%22S%22|activity_ID_PP.ID_PP:%2272%22|pagesize=10). However, the Packager only allows to accommodate "Ramo" in one classification.

![unnamed](https://user-images.githubusercontent.com/791678/34662391-eb97bb12-f458-11e7-801f-cd398a150f43.jpg)

> The consequence in the viewer is that the “Activity” tree map starts from “Modalidad” and it doesn’t allow to visualize the expenditure of unique Budget Programs by “Ramo”.

![unnamed 1](https://user-images.githubusercontent.com/791678/34662399-f8d80e94-f458-11e7-9bc4-c23982ad53c7.jpg)

![unnamed 2](https://user-images.githubusercontent.com/791678/34662407-0559a858-f459-11e7-8fc4-2e5df9ad7b46.jpg)

**Comment by pwalsh** [comment]
@vitorbaptista can you review this, and sync up with @akariv as needed, and then write an assessment of what, if anything, we can do to support this, or, note that this is a use case that simply will require direct usage of the API, rather than the vis layer.

**Comment by vitorbaptista** [comment]
@akariv Have you seen this issue before? Mexico has a single value that is used as part of two different dimensions, so would need a field with two `columnType`. This will require a change across everything that touches `columnType` (visualizations, schema editor, ...), but now is a good timing, as we'll touch these parts soon as part of moving to FDP 1.0. The question is how important/common this feature is.

If we don't have this feature, Mexico and others with the same issue would have to either:

1. Duplicate the column, using one `columnType` for each new field (e.g. `Ramo_Administrative` and `Ramo_Activity`)
1. Add the ID of `Ramo` to the fields that require it (`Modalidad` in the example), so they'll become "11", "12", "13", "21", "22", "23", where the first number is `ramo_id` and the second is `modalidad_id`

Neither is ideal.

**Comment by sglavoie** [comment]
**WONTFIX/CLEANUP**: Closing old issue without recent activity (>2.5 years) as part of #1478. Please feel free to re-open if this is still relevant.
